pkg/workflow/workload/pod/builder.go

```diff
@@ -344,9 +335,16 @@ func (m *Builder) ResolveInputResources() error {
 			})
 		}
 
+		// Get resource resolver for the given resource type. If the resource has resolver set, use it directly,
+		// otherwise get resolver from registered resource types.
+		resolver, err := common.GetResourceResolver(m.client, resource)
```

Comment:
If the client is for user clusters, then it can not get resource types.

Reply:
This client is used to be control cluster client.